### ISSUE TITLE
The database does not delete content synchronously？

### ISSUE THREAD

**Issue by pgy866** [OrchardCMS/OrchardCore#1238]
Delete the published content entries, the database still save the data, delete content is not synchronized.
`删除发布的内容条目，数据库内依旧保存数据，不同步删除内容。`

Do not know, is this design, or wrong?
`不知道是不是就是这样设计的，还是一个错误呢？`

Can  provide an action UI to completely delete deleted content? Keep clean database files.
`可以提供彻底删除已删除内容的操作UI吗？保留干净的数据库文件。`

![1](https://user-images.githubusercontent.com/28694676/32978779-f9dea692-cc83-11e7-9684-6e0fd4fe5276.png)

![2](https://user-images.githubusercontent.com/28694676/32978783-04daf1cc-cc84-11e7-91b9-02501e339890.png)

![3](https://user-images.githubusercontent.com/28694676/32978787-0a4fcdda-cc84-11e7-8b8f-a39e95c18e63.png)

![4](https://user-images.githubusercontent.com/28694676/32978789-0cfe1a8c-cc84-11e7-8600-3c0792e5deb4.png)

**Comment by sebastienros** [comment]
This is intentional, these are soft deletes. The records are kept but flagged as deleted, to preserve history. In the future we'll have a module to prune this data.

**Comment by EternityTeam** [comment]
@sebastienros Has this module been completed? In the latest development version, I found that logical deletion is still the default.

**Comment by sebastienros** [comment]
No, there is no such module. But I think there is an option on the content type to make the items non-versionable, in which case it won't store the previous versions.

**Comment by EternityTeam** [comment]
@sebastienros 
Yes, when I turn off the versionable option for content types, it does not generate multiple historical records after many modifications, but when I delete this content item, it sets the values of Latest and Published fields of the ContentItemIndex table to zero and the Id field gets a new value without deleting the index record, and the record in the Document table still exists.

I had delete the content item from UI，but...

![QQ截图20190524024809](https://user-images.githubusercontent.com/14150259/58278430-964eea80-7dce-11e9-9190-495e90a38445.png)
![QQ截图20190524024846](https://user-images.githubusercontent.com/14150259/58278431-96e78100-7dce-11e9-9da6-04770bc4e7a9.png)

**Comment by sebastienros** [comment]
Maybe an issue, if a content item is not versionable and deleted we could get rid of it.
